### PR TITLE
add(tur-on-device): new package blender

### PR DESCRIPTION
**Pull request by T-Dynamos**

### CODE REVIEW

**Comment by T-Dynamos** [comment]
```
This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
```
Sadness. How to free up space?

**Comment by licy183** [comment]
> ```
> This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
> ```
> 
> Sadness. How to free up space?

You can download the logs after the workflow is cancelled. Currently, the arm workflow is running under a self-hosted runner located in Shanghai China, and it may have a slow connection to Github. If the workflow x86/x86_64 has failed due to some reason, you can simply cancel all the workflows.

**Comment by T-Dynamos** [comment]
x86_64 process was killed by GitHub beacause of large space taken by it.

**Comment by T-Dynamos** [comment]
> simply cancel all the workflows.

I cannot do it as am not having write access to this repo.

**Comment by licy183** [comment]
> > simply cancel all the workflows.
> 
> I cannot do it as am not having write access to this repo. (https://docs.github.com/en/actions/managing-workflow-runs/canceling-a-workflow)

Emmm... I'm not very familiar with the Github Team Management. I've changed your role from Member to Owner so now you should have access to this repo.

**Comment by T-Dynamos** [comment]
@licy183 
Screenshot_20230319-141456_Chrome
How to fix it?

**Comment by licy183** [comment]
> How to fix it?

Try to change `built_tur_packages.txt` to `built_tur-on-device_packages.txt`

**Comment by T-Dynamos** [comment]
```
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package liblldb-6.0
E: Couldn't find any package by glob 'liblldb-6.0'
E: Unable to locate package libllvm6.0:amd64
E: Couldn't find any package by glob 'libllvm6.0'
```

**Comment by T-Dynamos** [comment]
@licy183 why termux docker doesn't have `libOpenSLES.so`
On Android
```
~$ ls /system/lib/ | grep OpenSLES                libOpenSLES.so
```

On termux docker returns nothing

**Comment by licy183** [comment]
> @licy183 why termux docker doesn't have `libOpenSLES.so` On Android
> 
> ```
> ~$ ls /system/lib/ | grep OpenSLES                
> libOpenSLES.so
> ```
> 
> On termux docker returns nothing

`libOpenSLES.so` and `libvulkan.so` will not be provided by `termux-docker` because they cannot work properly under this environment.

**Comment by T-Dynamos** [comment]
@licy183, why was aarch64 slow? Should we try QEMU now?

**Comment by licy183** [comment]
> @licy183, why was aarch64 slow? Should we try QEMU now?

It is hosted in China and may have connection issues to GitHub and `packages-cf.termux.dev`. Termux-docker under QEMU will not compile some C++ codes when I tested it locally so I don't want to use it.

**Comment by T-Dynamos** [comment]
@licy183 Can you check what is happening here? When I simply build it, action truncate build step with this error
```
This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
```
But when I use storing logs then uploading to GitHub artifacts method, x86 and x86_64 (aslo arm) build successfully but aarch64 action never stops and gets finally truncated after 6hrs.

Command I used for storing logs:
```bash
./common-files/build-termux-docker.sh login ./build-package.sh -I $packages 2>&1  | sudo tee build-${{ matrix.target_arch }}.log > /dev/null
```

**Comment by licy183** [comment]
> ```
> This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
> ```

This is not an error. After the action is finished, the log can be downloaded.

**Comment by licy183** [comment]
You can download the log here.
image

File: tur-on-device/blender/build.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+TERMUX_PKG_HOMEPAGE=https://www.blender.org
+TERMUX_PKG_DESCRIPTION="A fully integrated 3D graphics creation suite"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
```
Comment (licy183) [inline tur-on-device/blender/build.sh]: I think you should set yourself as the maintainer.

**Comment by licy183** [comment]
Thanks! I'll merge this now.

**Pull request merged**